### PR TITLE
Redis query

### PR DESCRIPTION
**Pull request by joeleba**
Large branch ahead
- Set up redis structure. Rmb to run `rake redis:setup`
- Lots of routing
- API for services, stops and sightings
- Back end logic for sighting
- Schema structure also changed. Please do the usual `rake db:drop db:create db:migrate`
- Seed for services A1 & A2. I'll add the rest later. Rmb to run `rake db:seed`

If there's anything you need me to clarify/change, just comment here.

### CODE REVIEW

File: src/app/controllers/api/v1/sightings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      def get_sighting
+        render json: Sighting.get_sighting(params)
+      end
+
+      # POST /api/v1/sightings
+      # Post a user report
+      #
+      # Params:
+      # sighting: {
+      #   user_id: ...
```
Comment (thamzhengyi) [inline src/app/controllers/api/v1/sightings_controller.rb:45]: Is this user id the facebook id?

Comment (joeleba) [inline src/app/controllers/api/v1/sightings_controller.rb:45]: user id. We don't use facebook id until much later (sharing and stuff :grinning: )

Comment (thamzhengyi) [inline src/app/controllers/api/v1/sightings_controller.rb:45]: Ah, so the front end is meant to cache the user id?

Comment (joeleba) [inline src/app/controllers/api/v1/sightings_controller.rb:45]: should be able to retrieve from rails session. I think it's in sessions_controller

File: src/app/models/sighting.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  # Return: { status: <failed/success> }
+  def self.post_sighting(params)
+    params = ActiveSupport::JSON.decode(params)
+    this_service = Service.find(params['service_id'])
+    this_stop = Stop.find(params['stop_id'])
+    this_user = User.find(params['user_id'])
+    status = 'failed'
+
+    User.transaction do
+      time_sheet = eval($redis.hget('tracking', this_service.id))
```
Comment (karen) [inline src/app/models/sighting.rb:52]: Is it a must to use `eval`?

Comment (joeleba) [inline src/app/models/sighting.rb:52]: ya I was trying to parse a string to a json object. JSON.decode just doesn't work here T.T

Comment (karen) [inline src/app/models/sighting.rb:52]: :ok_hand:

**Pull request merged**